beacon-chain/blockchain/receive_block.go

```diff
@@ -143,6 +144,8 @@ func (s *Service) ReceiveBlockBatch(ctx context.Context, blocks []*ethpb.SignedB
 	}
 
 	if err := s.VerifyWeakSubjectivityRoot(s.ctx); err != nil {
+		// log.Fatalf will prevent defer from being called
```

Comment:
I think the real problem is that we shouldn't be using log.Fatalf. This isn't a graceful shutdown and can cause other issues like corrupt data writes: https://github.com/prysmaticlabs/prysm/issues/7382

Reply:
I absolutely agree. We can discuss this during the OKRs meeting since I am about to take a look at logging.